include/cuco/detail/open_addressing_impl.cuh

```diff
@@ -0,0 +1,536 @@
+
+    size_type h_num_out;
+    CUCO_CUDA_TRY(
+      cudaMemcpyAsync(&h_num_out, d_num_out, sizeof(size_type), cudaMemcpyDeviceToHost, stream));
+    stream.synchronize();
+    std::allocator_traits<temp_allocator_type>::deallocate(
+      temp_allocator, reinterpret_cast<char*>(d_num_out), sizeof(size_type));
+    temp_allocator.deallocate(d_temp_storage, temp_storage_bytes);
+
+    return output_begin + h_num_out;
+  }
+
+  /**
+   * @brief Gets the number of elements in the container.
+   *
+   * @note This function synchronizes the given stream.
+   *
+   * @tparam Predicate Type of predicate indicating if the given slot is filled
+   *
+   * @param is_filled Predicate indicating if the given slot is filled
+   * @param stream CUDA stream used to get the number of inserted elements
+   *
+   * @return The number of elements in the container
+   */
+  template <typename Predicate>
+  [[nodiscard]] size_type size(Predicate const& is_filled, cuda_stream_ref stream) const noexcept
+  {
+    auto counter = detail::counter_storage<size_type, thread_scope, allocator_type>{allocator_};
+    counter.reset(stream);
+
+    auto const grid_size =
+      (storage_.num_windows() + detail::CUCO_DEFAULT_STRIDE * detail::CUCO_DEFAULT_BLOCK_SIZE - 1) /
+      (detail::CUCO_DEFAULT_STRIDE * detail::CUCO_DEFAULT_BLOCK_SIZE);
+
+    // TODO: custom kernel to be replaced by cub::DeviceReduce::Sum when cub version is bumped to
+    // v2.1.0
+    detail::size<detail::CUCO_DEFAULT_BLOCK_SIZE>
+      <<<grid_size, detail::CUCO_DEFAULT_BLOCK_SIZE, 0, stream>>>(
+        storage_.ref(), is_filled, counter.data());
+
+    return counter.load_to_host(stream);
+  }
+
+  /**
+   * @brief Gets the maximum number of elements the hash map can hold.
+   *
+   * @return The maximum number of elements the hash map can hold
+   */
+  [[nodiscard]] constexpr auto capacity() const noexcept { return storage_.capacity(); }
+
+  /**
+   * @brief Gets the sentinel value used to represent an empty key slot.
+   *
+   * @return The sentinel value used to represent an empty key slot
+   */
+  [[nodiscard]] constexpr key_type empty_key_sentinel() const noexcept
+  {
+    return empty_key_sentinel_;
+  }
+
+  /**
+   * @brief Gets the sentinel value used to represent an empty key slot.
+   *
+   * @return The sentinel value used to represent an empty key slot
+   */
+  [[nodiscard]] constexpr key_equal predicate() const noexcept { return predicate_; }
+
+  /**
+   * @brief Gets the sentinel value used to represent an empty key slot.
+   *
+   * @return The sentinel value used to represent an empty key slot
+   */
+  [[nodiscard]] constexpr probing_scheme_type probing_scheme() const noexcept
+  {
+    return probing_scheme_;
+  }
+
+  /**
+   * @brief Gets the sentinel value used to represent an empty key slot.
+   *
+   * @return The sentinel value used to represent an empty key slot
+   */
+  [[nodiscard]] constexpr allocator_type allocator() const noexcept { return allocator_; }
```

Comment:
```suggestion
  [[nodiscard]] constexpr allocator_type const& allocator() const noexcept { return allocator_; }
```

Reply:
discard this change for now to align with STL behavior